.github/workflows/test.yml

```diff
@@ -5,38 +5,76 @@ on: [push, pull_request]
 jobs:
   compilation-tests:
     runs-on: ubuntu-latest
+    env:
+      CI_COMMIT_MESSAGE: CI Formatting Auto Commit
+      CI_COMMIT_AUTHOR: ${{ github.event.repository.name }} CI
     steps:
       - uses: actions/checkout@v3
+        with:
+          # checkout the feature branch and be able to get the last commit of it
+          ref: ${{ github.event.pull_request.head.ref }}
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      # Set environment variables based on the last commit
+      - name: Set environment variable "commit-message"
+        run: echo "commit-message=$(git log -1 --pretty=format:'%s')" >> $GITHUB_ENV
+      - name: Display environment variable "commit-message"
+        run: echo "commit-message=${{ env.commit-message }}"
+      - name: Set environment variable "commit-author"
+        run: echo "commit-author=$(git log -1 --pretty=format:'%an')" >> $GITHUB_ENV
+      - name: Display environment variable "commit-author"
+        run: echo "commit-author=${{ env.commit-author }}"
+      - name: Set environment variable "is-auto-commit"
+        if: env.commit-message == env.CI_COMMIT_MESSAGE && env.commit-author == env.CI_COMMIT_AUTHOR
+        run: echo "is-auto-commit=true" >> $GITHUB_ENV
+      - name: Display environment variable "is-auto-commit"
+        run: echo "is-auto-commit=${{ env.is-auto-commit }}"
+
+      # Python setup
       - name: Set up Python 3.9
         uses: actions/setup-python@v4
+        if: env.is-auto-commit == false
```

Comment:
why can't we use the cache when the commit is the auto-commit?

If I understand correctly, when the `pull_request` event trigger this CI, then `is-auto-commit` will be false, which will in turn create another commit (in "Format all the cairo files and commit" step"). This new commit will also trigger the workflow, but will be detected as auto-commit and consequently skip all the tasks to go directly to the tests. Still it needs to have a working venv to perform those tests, right?

Reply:
Ah what I had in mind was to skip them when the auto-commit is pushed i.e. `is-auto-commit` is true so we don't need to test and build again, but I think it would be better to still do it and indeed we need a working venv to perform those steps ✅ let me play modify that

Reply:
auto format should work but I think that it's always a safe check to run test on the final code version

Reply:
Finally I took out the `is-auto-commit` check and just made it simpler. So on `push`, it runs `make format` and commit the diff, if any. In the pr, this commit is then added and could be reviewed as well